build.gradle

```diff
@@ -139,6 +139,20 @@ artifacts {
     archives javadocJar
 }
 
+import org.gradle.plugins.signing.Sign
+gradle.taskGraph.whenReady { taskGraph ->
+    if (taskGraph.allTasks.any { it instanceof Sign }) {
+        allprojects { ext."signing.keyId" = System.getenv('GPG_KEY_ID') }
+        allprojects { ext."signing.secretKeyRingFile" = System.getenv('GPG_KEY_LOCATION') }
+        allprojects { ext."signing.password" = System.getenv('GPG_PASSPHRASE') }
+    }
+    // Do not sign archives by default (a local build without gpg keyring should succeed)
+    if (taskGraph.allTasks.any { it.name == 'build' || it.name == 'assemble'  || it.name == 'publishToMavenLocal'}) {
```

Comment:
could you remove this and add:

```
ext.isReleaseVersion = getProjectProperty('TILEDB_JAVA_RELEASE').toBoolean()
tasks.withType(Sign) {
    onlyIf { isReleaseVersion }
}
```

I think it's effectively doing the same thing, add allprojects to the closure might not be correct.

Reply:
The release properties was what was not trigger the signing before, and I tested some changes to with no success. You can see for snapshots and for releases it's skipping the signing:
`Task :signMavenJavaPublication SKIPPED`

https://travis-ci.org/TileDB-Inc/TileDB-Java/jobs/574055440
https://travis-ci.org/TileDB-Inc/TileDB-Java/jobs/574370424